gnat2goto/driver/range_check.adb

```diff
@@ -108,11 +108,31 @@ package body Range_Check is
                   else UI_From_Int (2 ** (Type_Width - 1) - 1));
             begin
                return Make_Constant_Expr (Source_Location => No_Location,
-                                          I_Type          => Bound_Type,
+                                          I_Type          =>
+                                          Maybe_Double_Type_Width (Bound_Type),
                                           Range_Check     => False,
                                           Value =>
                                             Convert_Uint_To_Hex (Bound_Value,
-                                              Types.Pos (Type_Width)));
+                                              Types.Pos (Type_Width * 2)));
+            end;
+         when I_Floatbv_Type =>
+            declare
+               Smallest_Float : constant String := "C7EFFFFFE54DAFF8";
```

Comment:
What is the relationship between this string and the comment below?

Reply:
Equivalence. The number `-3.40282347E+39` is the small number representable in single-precision floats, and it's hex is `0xff7fffff`. The same number in double precision is `0xC7EFFFFFE54DAFF8`.

Reply:
Ummm... how did you come by that number?  The low bytes look a bit wonky to me.  Something I learned at great cost some years ago is that decimal is not a good way of working with binary floating-point.  It should only be used as the final step and for presentation.  Don't store floats as decimal and hope to get them back exactly, esp. crossing formats.

The most negative normal number in single precision floating-point is `-0x1.fffffep+127f` which has the bit pattern `0xff7fffff` and decimal value `-3.402823e+38f`.

The number in double is `-0x1.fffffep+127` (the last character is significant in C!) which then has bit pattern `0xc7efffffe0000000` and decimal value `-3.402823e+38`.

Reply:
I took the `0xff7fffff`, converted that to as 32-bit float to decimal (as above, but the power is 38, that's my typo). Then I converted that decimal to 64-bit double (via some online converter) and got `0xC7EFFFFFE54DAFF8`. This really is not my area so I can use your numbers if you prefer.

Reply:
That sounds consistent with what result you have got.  If you want the largest negative single precision float, as a double, then it is `0xc7efffffe0000000`.

Reply:
Ok, so I assume the largest positive single float as double is `0x47EFFFFFE0000000`?

Reply:
Yes!